apps/wallet/src/ui/app/staking/stake/StakingCard.tsx

```diff
@@ -0,0 +1,229 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import { getTransactionDigest, SUI_TYPE_ARG } from '@mysten/sui.js';
+import BigNumber from 'bignumber.js';
+import { Formik } from 'formik';
+import { useCallback, useMemo, useState } from 'react';
+import { Navigate, useNavigate, useSearchParams } from 'react-router-dom';
+
+import StakeForm from './StakeForm';
+import { ValidateDetailFormCard } from './ValidatorDetailCard';
+import { createValidationSchema } from './validation';
+import BottomMenuLayout, {
+    Content,
+    Menu,
+} from '_app/shared/bottom-menu-layout';
+import Button from '_app/shared/button';
+import Icon, { SuiIcons } from '_components/icon';
+import Loading from '_components/loading';
+import LoadingIndicator from '_components/loading/LoadingIndicator';
+import {
+    useAppSelector,
+    useAppDispatch,
+    useCoinDecimals,
+    useIndividualCoinMaxBalance,
+} from '_hooks';
+import {
+    accountAggregateBalancesSelector,
+    accountItemizedBalancesSelector,
+} from '_redux/slices/account';
+import { Coin, GAS_TYPE_ARG } from '_redux/slices/sui-objects/Coin';
+import { stakeTokens } from '_redux/slices/transactions';
+import { Text } from '_src/ui/app/shared/text';
+
+import type { SerializedError } from '@reduxjs/toolkit';
+import type { FormikHelpers } from 'formik';
+
+const initialValues = {
+    amount: '',
+};
+
+export type FormValues = typeof initialValues;
+
+function StakingCard() {
+    const coinType = GAS_TYPE_ARG;
+
+    const balances = useAppSelector(accountItemizedBalancesSelector);
+    const aggregateBalances = useAppSelector(accountAggregateBalancesSelector);
+    const coinBalance = useMemo(
+        () => (coinType && aggregateBalances[coinType]) || BigInt(0),
+        [coinType, aggregateBalances]
+    );
+    const [searchParams] = useSearchParams();
+    const validatorAddress = searchParams.get('address');
+    const unstake = !!searchParams.get('unstake');
```

Comment:
What's `unstake` in case of ...aUrl...?unstake

Reply:
Switch to the unstacking flow.